exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillSqlWorker.java

```diff
@@ -130,11 +132,18 @@ private static PhysicalPlan convertPlan(QueryContext context, String sql, Pointe
       logger.trace("There was an error during conversion into physical plan. " +
           "Will sync remote and local function registries if needed and retry " +
           "in case if issue was due to missing function implementation.", e);
-      // it is prohibited to retry query planning for ANALYZE statement since it changes
-      // query-level option values and will fail when rerunning with updated values
-      if (context.getFunctionRegistry().syncWithRemoteRegistry(
-              context.getDrillOperatorTable().getFunctionRegistryVersion())
-        && context.getSQLStatementType() != SqlStatementType.ANALYZE) {
+
+      int funcRegVer = context.getDrillOperatorTable().getFunctionRegistryVersion();
+      // We do not retry conversion if the error is a UserException of type RESOURCE
+      boolean isResourceErr = e instanceof UserException && ((UserException) e).getErrorType() == RESOURCE;
```

Comment:
@vvysotskyi when plugin auto disabling is switched on then this second attempt to convert the query fails with an error of "schema not found" because the plugin has been disabled. This error isn't as informative to the user as the original error raised when attempting to access the plugin so I didn't want this to be the error that they get back...

Reply:
Maybe after plugin auto disabling has happened I should set a flag somewhere like QueryContext to record that it has happened. Then this code doesn't need to try to tell what happened from the exception type, it can look at the flag and see that it should not try to sync the function registry and convert the query again because a needed plugin was broken and is now disabled?

Reply:
Yes, it would be a more reliable approach. Initially, I was looking for a way how to pass it properly, one of the options is to use the query option, but it will be tricky to set it in the place where plugins could be disabled with this feature.

Reply:
@vvysotskyi do you think that some new variable in UserException or UserExceptionContext which indicates that a plugin was disabled as a result of the underlying error would be preferable? This exception is visible in all of the needed places and seems like a pretty natural home for this information...